cycler.py

```diff
@@ -38,6 +38,30 @@
     {'color': 'b', 'linestyle': '-'}
     {'color': 'b', 'linestyle': '--'}
     {'color': 'b', 'linestyle': '-.'}
+
+RememberTheStyle
+================
+
+Helper class for mapping keys -> styles::
+
+   from cycler import RememberTheStyle, cycler
+
+   rts = RememberTheStyle((cycler('c', 'rgb') +
+                           cycler('ls', ['-', '--', ':']))
+
+   rts['cat']
```

Comment:
I agree with @pelson.  Invoking `y = x[key]` where this is the first time `key` has been seen looks like a mistake; it requires the reader to know too much about `x`.
Further, is the use case for this common enough to require a class?  How is it superior from the user's standpoint to assigning ```aardvark = next(my_cycler)``` as needed?  Or using the one line from the initializer, ```styles = dict(zip(names, my_cycler))```?
If you adopt Phil's version, the method could be called `named()`: 
```python
styles = NamedCycler(my_cycler)
plot(x, y, **styles.named('aardvark'))
```